### DISCUSSION TITLE
Make `Authorization Container` a Vendor container

### DISCUSSION THREAD

**Discussion by Mohammad-Alavi** [apiato/apiato#708] [Ideas]
I was thinking...
Maybe we can even extract the Authorization container and make it a vendor container. This way we can make the base Apiato even thinner!
The policies are a great way to enforce access control and roles & permissions are like a cherry on top and not needed always but can be brought in easily as a vendor container when needed!
I even though a bit about the implications of extracting it to a vendor container and:
1. I think it is easily doable
2. Nothing would change really, maybe just some setups if you need to use it for your project which is the case for every package anyone uses anyway.

My thinking behind this is to strip anything that is not necessary to start a project and let the use pull in the things that he/she really needs.
Remember when we had like 10 container in a fresh Apiato install? e.g. Setting, Payment. Now we have only 3. Do you feel the loss? I don't. And I can pull in e.g. Localization container when I need it as a vendor container. This will be the case with Authorization.

Your thoughts?

**Comment by wangdeer** [comment]
If put authorization in the vendor directory, where should I write custom authorization code if some projects require it? 
According to the standard, the code in the vendor directory will be overwritten during updates.

**Reply by Mohammad-Alavi** [reply]
@wangdeer Can you tell me an example of `custom authorization code`?

**Reply by wangdeer** [reply]
For example, I want to change the document annotation of this routing file to another language.

https://github.com/apiato/apiato/blob/896b9fdcf6bbbf5ef439ff3f99c68f17ab912402/app/Containers/AppSection/Authorization/UI/API/Routes/SyncUserRoles.v1.private.php
